lib/ability_extension_manager.rb

```diff
@@ -0,0 +1,19 @@
+class AbilityExtensionManager
```

Comment:
Can we add a short comment at the top?  About how the extensions build on each other and order matters?